### PR TITLE
Adds unit tests for RunIP and RunOVSDpctl functions present in util/ovs.go

### PR DESCRIPTION
**Pull request by sheetalsingala**
Signed-off-by: sheetalsingala <ssingala@redhat.com>

**- What this PR does and why is it needed**

Adds unit tests for RunIP and RunOVSDpctl functions present in util/ovs.go
**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/util/ovs_unit_test.go [hunk 1]
```diff
@@ -488,6 +489,60 @@ func TestRunOVSOfctl(t *testing.T) {
...
+	tests := []struct {
+		desc                    string
+		expectedErr             error
+		onRetArgsExecUtilsIface *onCallReturnArgs
+		onRetArgsKexecIface     *onCallReturnArgs
+	}{
+		{
+			desc:                    "negative: run `ovs-dpctl` command",
+			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
+			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string"}, []interface{}{nil, nil, fmt.Errorf("executable file not found in $PATH")}},
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:509]: L508: Calling `RunOVSDpctl` will never return `"executable file not found in $PATH"`. That error will be returned by SetExec() which ensures that the executable is at the right location. So, by the time we call RunOVSDpctl() we already know that the container has ovs-dpctl executable.

I know these are mocks, but for correctness return any error .. say "failed to execute ovs-dpctl command". Similarly, for all the other commands in this file.

Comment (vjayaramrh) [reply go-controller/pkg/util/ovs_unit_test.go:509]: @girishmg yes, you are right in your observation

File: go-controller/pkg/util/ovs_unit_test.go [hunk 2]
```diff
@@ -728,8 +783,8 @@ func TestRunOVNNbctlUnix(t *testing.T) {
 	}{
 		{
 			desc:                    "negative: run `ovn-nbctl` command with no env vars generated",
-			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
-			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string"}, []interface{}{nil, nil, fmt.Errorf("executable file not found in $PATH")}},
+			expectedErr:             fmt.Errorf("failed to execute ovs-nbctl command"),
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:786]: s/ovs-nbctl/ovn-nbctl/

File: go-controller/pkg/util/ovs_unit_test.go [hunk 3]
```diff
@@ -785,9 +840,9 @@ func TestRunOVNNbctlWithTimeout(t *testing.T) {
 	}{
 		{
 			desc:                    "negative: run `ovn-nbctl` command with timeout",
-			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
+			expectedErr:             fmt.Errorf("failed to execute ovs-nbctl command"),
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:843]: s/ovs-nbctl/ovn-nbctl/

File: go-controller/pkg/util/ovs_unit_test.go [hunk 4]
```diff
@@ -843,8 +898,8 @@ func TestRunOVNNbctl(t *testing.T) {
 	}{
 		{
 			desc:                    "negative: run `ovn-nbctl` command",
-			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
-			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string"}, []interface{}{nil, nil, fmt.Errorf("executable file not found in $PATH")}},
+			expectedErr:             fmt.Errorf("failed to execute ovs-nbctl command"),
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:901]: s/ovs-nbctl/ovn-nbctl/

File: go-controller/pkg/util/ovs_unit_test.go [hunk 5]
```diff
@@ -898,8 +953,8 @@ func TestRunOVNSbctlUnix(t *testing.T) {
 	}{
 		{
 			desc:                    "negative: run `ovn-sbctl` command with no env vars generated",
-			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
-			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string"}, []interface{}{nil, nil, fmt.Errorf("executable file not found in $PATH")}},
+			expectedErr:             fmt.Errorf("failed to execute ovs-sbctl command"),
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:956]: s/ovs-sbctl/ovn-sbctl/

File: go-controller/pkg/util/ovs_unit_test.go [hunk 6]
```diff
@@ -954,9 +1009,9 @@ func TestRunOVNSbctlWithTimeout(t *testing.T) {
 	}{
 		{
 			desc:                    "negative: run `ovn-sbctl` command with timeout",
-			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
+			expectedErr:             fmt.Errorf("failed to execute ovs-sbctl command"),
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:1012]: s/ovs-sbctl/ovn-sbctl/

File: go-controller/pkg/util/ovs_unit_test.go [hunk 7]
```diff
@@ -1012,8 +1067,8 @@ func TestRunOVNSbctl(t *testing.T) {
 	}{
 		{
 			desc:                    "negative: run `ovn-sbctl` command",
-			expectedErr:             fmt.Errorf("executable file not found in $PATH"),
-			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string"}, []interface{}{nil, nil, fmt.Errorf("executable file not found in $PATH")}},
+			expectedErr:             fmt.Errorf("failed to execute ovs-sbctl command"),
```
Comment (girishmg) [inline go-controller/pkg/util/ovs_unit_test.go:1070]: s/ovs-sbctl/ovn-sbctl/

File: go-controller/pkg/util/ovs_unit_test.go (hunk 2 above)
Comment (sheetalsingala) [reply go-controller/pkg/util/ovs_unit_test.go]: Will change, thank you

File: go-controller/pkg/util/ovs_unit_test.go [hunk 8]
```diff
@@ -488,6 +489,60 @@ func TestRunOVSOfctl(t *testing.T) {
...
+				ifaceCall.Arguments = append(ifaceCall.Arguments, mock.AnythingOfType(arg))
+			}
+			for _, ret := range tc.onRetArgsKexecIface.retArgList {
+				ifaceCall.ReturnArguments = append(ifaceCall.ReturnArguments, ret)
+			}
+			ifaceCall.Once()
+			_, _, e := RunOVSDpctl()
+
+			assert.Equal(t, e, tc.expectedErr)
+			mockExecRunner.AssertExpectations(t)
```
Comment (vjayaramrh) [inline go-controller/pkg/util/ovs_unit_test.go:541]: you will need `mockKexecIface.AssertExpectations(t)` as well

File: go-controller/pkg/util/ovs_unit_test.go [hunk 9]
```diff
@@ -1332,6 +1387,54 @@ func TestRunOVNSBAppCtl(t *testing.T) {
...
+				ifaceCall.Arguments = append(ifaceCall.Arguments, mock.AnythingOfType(arg))
+			}
+			for _, ret := range tc.onRetArgsKexecIface.retArgList {
+				ifaceCall.ReturnArguments = append(ifaceCall.ReturnArguments, ret)
+			}
+			ifaceCall.Once()
+			_, _, e := RunIP()
+
+			assert.Equal(t, e, tc.expectedErr)
+			mockExecRunner.AssertExpectations(t)
```
Comment (vjayaramrh) [inline go-controller/pkg/util/ovs_unit_test.go:1433]: you will need `mockKexecIface.AssertExpectations(t)` as well

**Pull request merged**